tools/code_format/check_format.py

```diff
@@ -122,15 +122,111 @@ def _normalize(
+    def excluded_prefixes(self):
+        return (
+            self.config.paths["excluded"] + tuple(self.args.add_excluded_prefixes)
+            if self.args.add_excluded_prefixes else self.config.paths["excluded"])
+
+    @property
+    def error_messages(self):
+        return []
+
+    @property
+    def operation_type(self):
+        return self.args.operation_type
+
+    @cached_property
+    def args(self):
+        parser = argparse.ArgumentParser(description="Check or fix file format.")
+        parser.add_argument(
+            "operation_type",
+            type=str,
+            choices=["check", "fix"],
+            help="specify if the run should 'check' or 'fix' format.")
+        parser.add_argument(
+            "target_path",
+            nargs="*",
+            default=["."],
+            help="specify the root directory for the script to recurse over. Default '.'.")
+        parser.add_argument("--path", default=".", help="specify the root path.")
+        parser.add_argument(
+            "--config_path",
+            default="./tools/code_format/config.yaml",
+            help="specify the config path. Default './tools/code_format/config.yaml'.")
+        parser.add_argument(
+            "--fail_on_diff",
+            action="store_true",
+            help="exit with failure if running fix produces changes.")
+        parser.add_argument(
+            "--add-excluded-prefixes", type=str, nargs="+", help="exclude additional prefixes.")
+        parser.add_argument(
+            "-j",
+            "--num-workers",
+            type=int,
+            default=multiprocessing.cpu_count(),
+            help="number of worker processes to use; defaults to one per core.")
+        parser.add_argument(
+            "--api-prefix", type=str, default="./api/", help="path of the API tree.")
+        parser.add_argument(
+            "--skip_envoy_build_rule_check",
+            action="store_true",
+            help="skip checking for '@envoy//' prefix in build rules.")
+        parser.add_argument(
+            "--namespace_check",
+            type=str,
+            nargs="?",
+            default="Envoy",
+            help="specify namespace check string. Default 'Envoy'.")
+        parser.add_argument(
+            "--namespace_check_excluded_paths",
+            type=str,
+            nargs="+",
+            default=[],
+            help="exclude paths from the namespace_check.")
+        parser.add_argument(
+            "--build_fixer_check_excluded_paths",
+            type=str,
+            nargs="+",
+            default=[],
+            help="exclude paths from envoy_build_fixer check.")
+        parser.add_argument(
+            "--bazel_tools_check_excluded_paths",
+            type=str,
+            nargs="+",
+            default=[],
+            help="exclude paths from bazel_tools check.")
+        parser.add_argument(
+            "--clang_format_path", type=str, help="Path to clang-format executable.")
```

Comment:
Consider taking it from the env variable as a default value (doesn't have to be in this PR).

Reply:
apologies - misread - i think i have added this here incorrectly ill check

Reply:
i removed from this PR it was an artefact of the next